### ISSUE TITLE
[tracking issue] RPC tests are flaky

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#29117]
```
======================================================================
Nov 01 22:11:17 ERROR: test_call_method_on_rref (__main__.RpcTestWithFork)
Nov 01 22:11:17 ----------------------------------------------------------------------
Nov 01 22:11:17 Traceback (most recent call last):
Nov 01 22:11:17   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 130, in wrapper
Nov 01 22:11:17     self._join_processes(fn)
Nov 01 22:11:17   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 211, in _join_processes
Nov 01 22:11:17     self._check_return_codes(elapsed_time)
Nov 01 22:11:17   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 230, in _check_return_codes
Nov 01 22:11:17     raise RuntimeError('Process {} terminated or timed out after {} seconds'.format(i, elapsed_time))
Nov 01 22:11:17 RuntimeError: Process 0 terminated or timed out after 400.1577408313751 seconds
Nov 01 22:11:17 
Nov 01 22:11:17 ----------------------------------------------------------------------
```

cc @ezyang @gchanan @zou3519 @jerryzh168 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528

**Comment by ezyang** [comment]
https://app.circleci.com/jobs/github/pytorch/pytorch/3442436

**Comment by mruberry** [comment]
This may only be for ASAN.

**Comment by mruberry** [comment]
@aazzolini Who wrote this test recently.

Does this test need a retry mechanism? Should it be disabled on ASAN builds?

**Comment by aazzolini** [comment]
This test shouldn't be flaky neither take 400+ seconds as it is very simple. I'm investigating.

**Comment by pietern** [comment]
https://circleci.com/gh/pytorch/pytorch/3467580

**Comment by suo** [comment]
https://l.prod.workplace.com/l.php?u=https%3A%2F%2Fapp.circleci.com%2Fjobs%2Fgithub%2Fpytorch%2Fpytorch%2F3463855&h=AT0ieijWPtblu4dOMQcdK01ePQK3xCHc3mUh_U2ZCqvR4DWSZdjHrzP9jpbA2uUd6RkH5ifRSpwEMVFo0RDioThHQIcvcbd4c8KYRGTHE7rHMGgzt8N1PX_HaVFIgDrfDMYrkFatPvCv7Sno_uBdVQFIhWiiwaXuftAVaQ

**Comment by suo** [comment]
https://app.circleci.com/jobs/github/pytorch/pytorch/3503307

**Comment by mrshenli** [comment]
Fixes are on the way #29069  #29306. Flakiness address in these two PRs have broad impact on almost all rpc/rref tests. Hopefully, this could give us a relief.

**Comment by suo** [comment]
https://app.circleci.com/jobs/github/pytorch/pytorch/3518795

**Comment by suo** [comment]
https://app.circleci.com/jobs/github/pytorch/pytorch/3521041

**Comment by suo** [comment]
https://app.circleci.com/jobs/github/pytorch/pytorch/3529088
(let me know if this is not useful)

**Comment by mrshenli** [comment]
closed by #29827
